### PR TITLE
Remove unnecessary prefixes from OperationIDs in OpenAPI spec

### PR DESCRIPTION
**Pull request by mbohlool**
When we moved to one spec (to rule them all #35388), we added prefix to all Operation IDs in OpenAPI spec (to be precise, the prefix is added after the verb, for example a listPod converted to listCoreV1Pods). These prefixes are not necessary for all GroupVersions and OperationIDs. Some Operation IDs such as getAPIResources defined for more than one GroupVersions and some GroupVersions such as extensions_v1beta1 defined duplicate operation IDs because their operation are release in v1 and still exists in v1beta1. This PR will try to minimize adding prefixes to have a more readable operation IDs. These IDs will be used as method names in clients so it is important to keep them as short as possible.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36002)

### CODE REVIEW

**Comment by nikhiljindal** [comment]
cc @kubernetes/sig-api-machinery

File: pkg/apiserver/openapi/openapi.go [hunk 1]
```diff
@@ -29,7 +29,21 @@ import (
...
+var forcePrefixForOpIDs = map[string]bool{
+	"getAPIResources": true,
+	"getAPIGroup":     true,
+	"getAPIVersions":  true,
+}
+
+// forcePrefixForTags contains list of tags which their Operation IDs will get a prefix (usually GroupVersion) to prevent duplicates.
+// It is preferred to put alpha/beta (e.g., *_v1beta1) tags in this list rather than released tags (e.g. CoreV1).
+// Tags are usually lowercase group or group_version
+var forcePrefixForTags = map[string]bool{
```
Comment (smarterclayton) [inline pkg/apiserver/openapi/openapi.go]: This doesn't belong in this package - this package should not know anything about specific versions.  You must pass this in.

Comment (mbohlool) [inline pkg/apiserver/openapi/openapi.go]: This file has already had some version specific code. Should I move it completely to master package?

Comment (ncdc) [inline pkg/apiserver/openapi/openapi.go]: I'll take a look. cc @deads2k @sttts @liggitt

Comment (mbohlool) [inline pkg/apiserver/openapi/openapi.go]: @ncdc @smarterclayton Thanks for looking into this. I appreciate it if we can get this PR in 1.5. 

The point of ToValidOperationID method is to customize OperaionIDs and in my opinion any API server should customize its own. Almost all of the code in that method is specific to the way we setup k8s API server. Things like api->apis/core, apis prefix, removing .k8s.io suffix, fixing Scale OpID) are all k8s specific. That is why I suggest we move this function to master. ToValidOperationID can be reused and I can move it to a common package.

**Comment by mbohlool** [comment]
I've made both ForcedPrefixForTag and ForcedPrefixForOpId list a parameter and feed them from the outside of apiserver package. Please take another look.

**Comment by smarterclayton** [comment]
changes look fine

**Comment by smarterclayton** [comment]
apply label when green

**Comment by mbohlool** [comment]
@k8s-bot cvm gce e2e test this issue #36153

**Comment by mbohlool** [comment]
@k8s-bot cvm gce e2e test this issue #36153

**Comment by mbohlool** [comment]
@smarterclayton Thanks.

**Comment by mbohlool** [comment]
@k8s-bot cvm gce e2e test this issue #36153

**Comment by mbohlool** [comment]
@k8s-bot gce e2e test this issue #36153

**Comment by mbohlool** [comment]
@k8s-bot kubemark e2e test this issue #IGNORE

**Comment by mbohlool** [comment]
After a discussion with @smarterclayton, he convinced me that consistency of what we already have is more important than the connivence this will give us. Also there is a risk of conflicting Operation IDs with this change that will prevent API server to start. I won't take that risk now and agree to consistency. The convenience may be achievable on the client-generation side. I will investigate that.

**Pull request closed without merging**